File: back_end/object/Post.go
package object

type Post struct {
	OrmModal
	UserID      uint      `json:"userId"`
	User        User      `gorm:"foreignKey:UserID" json:"user"`
	Title       string    `gorm:"type:varchar(100)" json:"title"`
	Description string    `gorm:"type:text" json:"description"`
	Content     string    `gorm:"type:text" json:"content"`
	Comments    []Comment `gorm:"foreignKey:PostID" json:"comments"`
	Likes       int       `json:"likes"`
	Views       int       `json:"views"`
	Favorites   int       `json:"favorites"`
	Forwards    int       `json:"forwards"`
}

File: fonted_end/src/interface/Post.ts
import {IUser} from "./User.ts";

export interface IPost {
    id: number;
    title: string;
    description: string;
    content: string;
    avatar: string;
    href: string;
    author: string;
    date: string;
    comments: IComment[];
    //点赞
    likes: number;
    //浏览
    views: number;
    //收藏
    favorites: number;
    //转发
    forwards: number;
}

export interface IComment {
    id: number;
    author: IUser;
    content: string;
    date: string;
}
